src/main/java/doge/command/DeleteCommand.java

```diff
@@ -0,0 +1,57 @@
+package doge.command;
+
+import doge.Storage;
+import doge.TaskList;
+import doge.Ui;
+import doge.exception.DogeException;
+
+/**
+ * Represents the "delete" command. Doge bot will delete a specified task from the TaskList.
+ */
+public class DeleteCommand extends Command {
+
+    /**
+     * Constructor for class DeleteCommand
+     *
+     * @param s the task no. that will be deleted
+     */
+    public DeleteCommand(String s) {
+        super(s);
+    }
+
+    /**
+     * Executes the "delete" command. It deletes the specified task from the TaskList.
+     *
+     * @param tasks {@inheritDoc}
+     * @param ui {@inheritDoc}
+     * @param storage {@inheritDoc}
+     * @throws DogeException if it fails to delete the specified task or user failed to provide the task number
+     */
+    public void execute(TaskList tasks, Ui ui, Storage storage) throws DogeException {
+        if (this.details.isEmpty()) {
+            throw new DogeException("How am I suppose to delete something without knowing which task?");
+        }
+
+        int pos;
+        try {
+            pos = Integer.parseInt(this.details);
+        } catch (NumberFormatException e) {
+            throw new DogeException("Are you incapable of understanding what's an integer? Input valid task no.");
+        }
+
+        if (pos > tasks.size() || pos < 0) {
```

Comment:
It seems that you are doing this position check in other subclass of the Command class too (for eg in FindCommand too), Do you think it will ne better if you abstract this out as a method in the Command class instead?

Reply:
That seems to be a good idea to further categorize the `Command` class as commands like delete, mark, unmark requires the same integer input at `pos`. But adding to the main Command class would be troublesome as other commands do not really have the same need. So maybe creating more subclasses of Command would be a better approach?